File: demos/usage-grid-layout.py
"""
Original Demo: http://js.cytoscape.org/demos/grid-layout/
"""
import json

import dash
from dash import html

import dash_cytoscape as cyto

app = dash.Dash(__name__)
server = app.server


# Load Data
with open("data/grid-layout/data.json", "r", encoding="utf-8") as f:
    elements = json.loads(f.read())

# App
app.layout = html.Div(
    [
        cyto.Cytoscape(
            id="cytoscape",
            elements=elements,
            layout={"name": "grid"},
            stylesheet=[
                {
                    "selector": "node",
                    "style": {"height": 20, "width": 20, "background-color": "#18e018"},
                },
                {
                    "selector": "edge",
                    "style": {
                        "curve-style": "haystack",
                        "haystack-radius": 0,
                        "width": 5,
                        "opacity": 0.5,
                        "line-color": "#a2efa2",
                    },
                },
            ],
            style={
                "width": "100%",
                "height": "100%",
                "position": "absolute",
                "left": 0,
                "top": 0,
                "z-index": 999,
            },
        )
    ]
)

if __name__ == "__main__":
    app.run(debug=True)

File: demosR/usage-grid-layout.R
# original demo: http://js.cytoscape.org/demos/grid-layout/

library(dash)
library(dashCytoscape)
library(jsonlite)

app <- Dash$new()

# load data
elements <- fromJSON("demos/data/grid-layout/data.json")

# define app layout
app$layout(
  htmlDiv(
    list(
      cytoCytoscape(
        id = 'cytoscape',
        elements = elements,
        layout = list('name' = 'grid'),
        stylesheet = list(
          list(
            'selector' = 'node',
            'style' = list(
              'height' = 20,
              'width' = 20,
              'background-color' = '#18e018'
            )
          ),
          list(
            'selector' = 'edge',
            'style' = list(
              'curve-style' = 'haystack',
              'haystack-radius' = 0,
              'width' = 5,
              'opacity' = 0.5,
              'line-color' = '#a2efa2'
            )
          )
        ),
        style = list(
          'width' = '100%',
          'height' = '100%',
          'position' = 'absolute',
          'left' = 0,
          'top' = 0,
          'z-index' = 999
        )
      )
    )
  )
)

app$run_server(debug = TRUE)
